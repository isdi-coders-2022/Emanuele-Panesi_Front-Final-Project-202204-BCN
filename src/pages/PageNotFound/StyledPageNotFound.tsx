import styled from "styled-components";

const StyledPageNotFound = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: "Roboto";

  span {
    font-size: 25px;
    font-weight: 400;
    text-align: center;
    letter-spacing: 3px;
  }

  .content {
    min-height: calc(100vh - 70px);
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  img {
    width: 200px;
    height: 200px;
  }

  .backhome-button {
    width: 210px;
    height: 80px;
    border-radius: 4rem;
    color: rgb(38, 50, 56);
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
    color: black;
    border: none;
    cursor: pointer;
    letter-spacing: 2px;
    font-weight: 300;
    font-size: 25px;
    border: 1px solid rgb(38, 50, 56);
    margin-top: 150px;
  }
`;

export default StyledPageNotFound;
