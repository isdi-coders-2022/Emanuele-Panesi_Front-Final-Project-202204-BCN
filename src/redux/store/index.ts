import { configureStore } from "@reduxjs/toolkit";
import paginationReducer from "../features/paginationSlice";
import pizzeriaReducer from "../features/pizzeriaSlice";
import pizzeriasReducer from "../features/pizzeriasSlice";
import userReducer from "../features/userSlice";
import uiReducer from "../thunks/uiSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    pizzerias: pizzeriasReducer,
    pizzeria: pizzeriaReducer,
    pages: paginationReducer,
    ui: uiReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
