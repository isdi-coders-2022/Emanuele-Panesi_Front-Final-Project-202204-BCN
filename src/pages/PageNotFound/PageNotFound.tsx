import { useNavigate } from "react-router-dom";
import StyledPageNotFound from "./StyledPageNotFound";

const PageNotFound = () => {
  const navigate = useNavigate();

  const navigateToHome = () => {
    navigate("/home");
  };

  return (
    <>
      <StyledPageNotFound>
        <div className="content">
          <span className="message">Sorry</span>
          <span className="message">Page not found</span>
          <img src="/images/pizza-pagenotfound.png" alt="Pizza slice" />
          <button className="backhome-button" onClick={navigateToHome}>
            BACK TO HOMEPAGE
          </button>
        </div>
      </StyledPageNotFound>
    </>
  );
};

export default PageNotFound;
