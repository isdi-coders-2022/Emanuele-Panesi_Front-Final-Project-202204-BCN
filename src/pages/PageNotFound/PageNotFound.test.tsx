import { Provider } from "react-redux";
import store from "../../redux/store";

import { BrowserRouter } from "react-router-dom";
import reactTestRenderer from "react-test-renderer";
import PageNotFound from "./PageNotFound";

describe("Given a PageNotFound Page", () => {
  describe("When it's instantiated", () => {
    test("Then it should render an image and a button", () => {
      const pageNotFound = reactTestRenderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <PageNotFound />
          </Provider>
        </BrowserRouter>
      ).toJSON;

      expect(pageNotFound).toMatchSnapshot();
    });
  });
});
