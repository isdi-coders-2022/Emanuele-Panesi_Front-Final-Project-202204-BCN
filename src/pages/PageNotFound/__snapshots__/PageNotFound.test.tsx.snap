// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PageNotFound Page When it's instantiated Then it should render an image and a button 1`] = `[Function]`;
